### DISCUSSION TITLE
Feature Request: Error Object

### DISCUSSION THREAD

**Discussion by nahtnam** [next-safe-action/next-safe-action#125] [Ideas]
I like how the error handling is set up: https://next-safe-action.dev/docs/safe-action-client/custom-server-error-handling

But I think it would be nice if the library also exposed something like `TRPCError` where we can throw things like unauthorized, 400, etc in a standard format

I saw here: https://github.com/TheEdoRan/next-safe-action/discussions/88, an example with `class ActionError extends Error {}`. Would be nice if it was integrated with the library automatically

**Comment by TheEdoRan** [comment]
Server Actions always return status HTTP 200 OK, and it's not customizable. I think that allowing users to create custom errors is good enough, since HTTP statuses are not used by underlying React Actions.
